proc_netstat.go

```diff
@@ -205,19 +194,28 @@ func parseNetstat(r io.Reader, fileName string) (map[string]map[string]float64,
 		valueParts := strings.Split(scanner.Text(), " ")
 		// Remove trailing :.
 		protocol := nameParts[0][:len(nameParts[0])-1]
-		netStats[protocol] = map[string]float64{}
 		if len(nameParts) != len(valueParts) {
-			return nil, fmt.Errorf("mismatch field count mismatch in %s: %s",
+			return ProcNetstat{}, fmt.Errorf("mismatch field count mismatch in %s: %s",
 				fileName, protocol)
 		}
 		for i := 1; i < len(nameParts); i++ {
-			var err error
-			netStats[protocol][nameParts[i]], err = strconv.ParseFloat(valueParts[i], 64)
+			value, err := strconv.ParseFloat(valueParts[i], 64)
 			if err != nil {
-				return nil, err
+				return procNetstat, err
+			}
+			key := nameParts[i]
+
+			var s interface{}
+			if protocol == "TcpExt" {
+				s = &procNetstat.TcpExt
 			}
+			if protocol == "IpExt" {
+				s = &procNetstat.IpExt
+			}
+			structValue := reflect.ValueOf(s).Elem()
```

Comment:
We try to avoid using reflect. I'd use a switch/case statement to map the values. A bit verbose but should be fine.

Reply:
@discordianfish Sure i will do, but just from curiosity why we avoid `reflect`?